daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -811,3 +810,51 @@ object Transaction {
   }
 
 }
+
+/** An explicitly-disclosed contract that has been used during command interpretation
+  * and enriched with additional contract metadata.
+  *
+  * @param createEvent the create event of the contract
+  * @param createdAt ledger effective time of the transaction that created the contract
+  * @param driverMetadata opaque bytestring used by the underlying ledger implementation
+  */
+final case class DisclosedEvent(
```

Comment:
I would keep the `DisclosedContract` and `disclosedContracts` name for representing this bunch at least at interfaces (LAPI and WriteService):
1. Content-wise, it's the same. You just simplified stuff, no?
2. Involves less change in this codebase (all the renames)
3. If we rename, it becomes more ambiguous wrt to the docs that would need to be adapted as well (see `WriteService.submitTransaction`)

Wdyt?

Reply:
This is a air question @tudor-da.

In general, I am concerned about trying to name thing well. In general I do not really like adding a "vague" prefix (like Processed or Enriched) in front of name to distinguish close but different data structure. It is better than nothing but can we do better ?

Now we already have a data structure called CreateEvent that we expose to the ledger API (it is not some kind of obscure concept) that contains almost the same info as this one (we are just missing the views which are computed lazily), that I think we should reuse terminology. Afterthought DisclosedCreateEvent is probably better than DisclosedEvent.

In my mind, DisclosedContract is the input (what is send with the command) and DisclosedCreateEvent is the output (the engine reconstruct it from the input).

The purpose of this PR was two folds, (1) simplified stuff and (2) improve ramming.

I probably should have make a PR to handling each one separately, but this probably means two breaking changes for canton.

I personally do not buy the argument to favor minimal change over naming improvement, specially for a feature which is not stable.

If you do not agree with me on (2), I will simply revert back the name to ProcessedDisclosedContract without further discussion, if not please point to me the documentation that needs to be changed.

Reply:
> I personally do not buy the argument to favor minimal changes over naming improvement, especially for a feature which is not stable.

I don't see the change from `DisclosedContract` to `DisclosedEvent` as an improvement but maybe I'm missing something from your context. On the point, maybe my wording could have suggested not to do it because it's cumbersome. So rephrasing as I don't see from my pov justification in doing it this way. `DisclosedEvent` sounds too generic to me as I mentioned in another comment. 

> If you do not agree with the naming improvement is substantial enough, I will simply revert back the name to ProcessedDisclosedContract without further discussion (nor resentment 😉), if not please point to me the documentation that needs to be changed.

I had no intention of nitpicking here, especially about naming which is hard and subjective! But I did see value in my comments. Anyway, happy to disagree on this 👍

Reply:
5c1908a78a7568cc026391a13de32ab192e20794